### DISCUSSION TITLE
Meetings 2022-02-10

### DISCUSSION THREAD

**Discussion by jongfeel** [hongsam14/myCPU#30] [General]
## Pull request guide

- Pull request의 단위를 작게 가져가고 빠르게 merge 진행
- Reviewer를 저로 설정, 1~2일 안에 merge 진행할 수 있고 코멘드 드립니다
- 리팩토링(로직이랑 상관 없는 띄어쓰기 수정 commit) 내용은 모아서 따로 pull request로 만드는 작전도 좋음

## Working problems

- MVC를 분리하여 진행하는 간단한 방법
  - MonoBehaviour를 상속받는 클래스 => controller
  - 그냥 C# 클래스 => model
- 가비지 콜렉터에 대해 공부하는 계기, 코드를 짜면서 메모리 관리를 하다가 알게 됨.
- delegate의 활용은 Action, Func로 대체해서 사용
- operator overloading 필요하다면 구현할 수 있음
- windowCache 창에 대한 관리 => List로 관리하면 Dictionary 안써도 될 듯 합니다.
- [field:SerializedField] attribue로 property를 inspector에 노출 가능
- inspector를 잠그고 windowCache를 처리하는 방향으로도 가능

**Comment by hongsam14** [comment]
주말동안 이것저것 작업을 해보았습니다.
1. [field:SerializedField]를 사용했을때, inspector가 노출이 되는데, 이 값에 접근을 할 방법을 찾고 싶었지만 현재 실패한 상태...
2. 콜백을 하지않고 inspector를 잠그고 object를 받아오는 방법이 있을 것 같아 시도해보았지만, 콜백으로 가져오는 방법이 훨씬 자연스러운 것 같다...
3. static 메모리를 안쓰는 아이디어를 떠올리려고 노력중...
